modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/throttling/SpeedBasedMemoryConsumptionThrottlingStrategy.java

```diff
@@ -496,5 +506,35 @@ void reset() {
     void finish() {
         cpWriteSpeed.closeInterval();
         threadIds.clear();
+
+        updateWriteVsFsyncCoefficient();
+    }
+
+    private void updateWriteVsFsyncCoefficient() {
+        CheckpointProgress progress = cpProgress.get();
+        assert progress != null;
+
+        if (progress.currentCheckpointPagesCount() == 0) {
+            return;
+        }
+
+        long pagesWriteTimeMillis = progress.getPagesWriteTimeMillis();
+        long fsyncTimeMillis = progress.getFsyncTimeMillis();
+
+        double coefficient = ((double) pagesWriteTimeMillis) / (pagesWriteTimeMillis + fsyncTimeMillis);
+        if (isNaN(coefficient)) {
+            return;
+        }
+
+        double newCoefficient = writeVsFsyncCoefficient * 0.85 + coefficient * 0.15;
+
+        // Put it within reasonable bounds just in case.
```

Comment:
What do you mean by "reasonable" bounds?

Reply:
I don't want any of these values to be too close to zero

Reply:
Why?

Reply:
Too afraid of some weird artifacts. I think that 0.9 upper bound can be ignored, but the lower bound is mandatory, otherwise we'll get huge stalls on malfunctioning or otherwise strangely behaving storages